python/pylibcudf/pylibcudf/libcudf/join.pxd

```diff
@@ -19,7 +19,7 @@ from pylibcudf.libcudf.utilities.span cimport device_span
 
 ctypedef unique_ptr[device_uvector[size_type]] gather_map_type
 ctypedef pair[gather_map_type, gather_map_type] gather_map_pair_type
-ctypedef optional[pair[size_t, device_span[const size_type]]] output_size_data_type
+ctypedef optional[size_t] output_size_data_type
```

Comment:
Let's remove this typedef and use `optional[size_t]` directly in the forward declarations.

Can we also please rename the parameter from `output_size_data` to `output_size` (since it no longer takes the data as well).

So, for example:

```
cdef gather_map_pair_type mixed_left_join(
   ...
   optional[size_t] output_size,
   ...
)
```